x-pack/legacy/plugins/actions/server/builtin_action_types/servicenow.ts

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { curry } from 'lodash';
+import { i18n } from '@kbn/i18n';
+import { schema, TypeOf } from '@kbn/config-schema';
+import {
+  ActionType,
+  ActionTypeExecutorOptions,
+  ActionTypeExecutorResult,
+  ExecutorType,
+} from '../types';
+import { ActionsConfigurationUtilities } from '../actions_config';
+import { postServiceNow } from './lib/post_servicenow';
+
+// config definition
+export type ConfigType = TypeOf<typeof ConfigSchema>;
+
+const ConfigSchemaProps = {
+  apiUrl: schema.string(),
+};
+
+const ConfigSchema = schema.object(ConfigSchemaProps);
+
+function validateConfig(
+  configurationUtilities: ActionsConfigurationUtilities,
+  configObject: ConfigType
+) {
+  if (configObject.apiUrl == null) {
+    return i18n.translate('xpack.actions.builtin.servicenow.servicenowApiNullError', {
+      defaultMessage: 'ServiceNow [apiUrl] is required',
+    });
+  }
+  try {
+    configurationUtilities.ensureWhitelistedUri(configObject.apiUrl);
+  } catch (whitelistError) {
+    return i18n.translate('xpack.actions.builtin.servicenow.servicenowApiWhitelistError', {
+      defaultMessage: 'error configuring servicenow action: {message}',
+      values: {
+        message: whitelistError.message,
+      },
+    });
+  }
+}
+// secrets definition
+export type SecretsType = TypeOf<typeof SecretsSchema>;
+const SecretsSchemaProps = {
+  password: schema.string(),
+  username: schema.string(),
+};
+
+const SecretsSchema = schema.object(SecretsSchemaProps);
+
+function validateSecrets(
+  configurationUtilities: ActionsConfigurationUtilities,
+  secrets: SecretsType
+) {
+  if (secrets.username == null) {
+    return i18n.translate('xpack.actions.builtin.servicenow.servicenowApiUserError', {
+      defaultMessage: 'error configuring servicenow action: no secrets [username] provided',
+    });
+  }
+  if (secrets.password == null) {
+    return i18n.translate('xpack.actions.builtin.servicenow.servicenowApiPasswordError', {
+      defaultMessage: 'error configuring servicenow action: no secrets [password] provided',
+    });
+  }
+}
+
+// params definition
+
+export type ParamsType = TypeOf<typeof ParamsSchema>;
+
+const ParamsSchema = schema.object({
+  comments: schema.string(),
+  short_description: schema.string(),
+});
```

Comment:
I had a look at [prometheus](https://github.com/FXinnovation/alertmanager-webhook-servicenow#configuration)  and [elastalert](https://elastalert.readthedocs.io/en/latest/ruletypes.html#servicenow) for comparison, and they include several other incident fields:

- assignment_group
- category
- subcategory
- cmdb_ci 
- caller_id
- urgency
- description
- impact

Would it make sense to include any of these other fields? They all appear to be optional.